src/Ryujinx.HLE/HOS/Services/Fs/FileSystemProxy/IFileSystem.cs

```diff
@@ -149,7 +150,15 @@ public ResultCode OpenDirectory(ServiceCtx context)
         // Commit()
         public ResultCode Commit(ServiceCtx context)
         {
-            return (ResultCode)_fileSystem.Get.Commit().Value;
+            Result result = _fileSystem.Get.Commit();
+            ResultCode resultCode = (ResultCode)result.Value;
+
+            if (result.IsFailure() && resultCode == ResultCode.PathAlreadyInUse)
```

Comment:
```suggestion
            if (resultCode == ResultCode.PathAlreadyInUse)
```
I don't think checking if the result is an error is necessary since you already compare with a specific error code anyway.